integrations/cdi/eclipselink-cdi/src/main/java/io/helidon/integrations/cdi/eclipselink/CDISEPlatform.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.integrations.cdi.eclipselink;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Objects;
+import java.util.concurrent.Executor;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Qualifier;
+import javax.management.MBeanServer;
+import javax.transaction.TransactionManager;
+
+import org.eclipse.persistence.platform.server.JMXServerPlatformBase;
+import org.eclipse.persistence.platform.server.ServerPlatformBase; // for javadoc only
+import org.eclipse.persistence.sessions.DatabaseSession;
+import org.eclipse.persistence.sessions.JNDIConnector;
+import org.eclipse.persistence.transaction.JTATransactionController;
+
+/**
+ * A {@link JMXServerPlatformBase} that arranges things such that CDI,
+ * not JNDI, will be used to acquire a {@link TransactionManager} and
+ * {@link MBeanServer}.
+ *
+ * <p>Most users will not use this class directly, but will supply its
+ * fully-qualified name as the value of the <a
+ * href="https://www.eclipse.org/eclipselink/documentation/2.7/jpa/extensions/persistenceproperties_ref.htm#target-server">{@code
+ * eclipselink.target-server} Eclipselink JPA extension property</a>
+ * in a <a
+ * href="https://javaee.github.io/tutorial/persistence-intro004.html#persistence-units">{@code
+ * META-INF/persistence.xml} file</a>.</p>
+ *
+ * <p>For example:</p>
+ *
+ * <blockquote><pre>&lt;property name="eclipselink.target-server"
+ *          value="io.helidon.integrations.cdi.eclipselink.CDISEPlatform"/&gt;</pre></blockquote>
+ *
+ * @author <a href="mailto:laird.nelson@oracle.com">Laird Nelson</a>
+ *
+ * @see #getExternalTransactionControllerClass()
+ */
+public class CDISEPlatform extends JMXServerPlatformBase {
+
+
+    /*
+     * Constructors.
+     */
+
+
+    /**
+     * Creates a {@link CDISEPlatform}.
+     *
+     * @param session the {@link DatabaseSession} this platform will
+     * wrap; must not be {@code null}
+     *
+     * @see JMXServerPlatformBase#JMXServerPlatformBase(DatabaseSession)
+     */
+    public CDISEPlatform(final DatabaseSession session) {
```

Comment:
Please do not use `final` for parameters or variables.

Reply:
Is the public constructor required because it is instantiated by CDI? If not, use a static factory method and the constructor should be private.

Reply:
It is instantiated by EclipseLink.  Yes, it must be `public`.

Reply:
Regarding `final` variables and parameters: checkstyle passed as-is.  What would be the motivation for forbidding their usage?

Reply:
They are not used in other Helidon code - consistency is the only reason.
Also such declaration is mostly useless - the only reason for using them may be to prevent yourself from assigning a value to a parameter (which can be caught by IDE validations)

Reply:
I have a very strong preference for final myself, everywhere possible: it absolutely clarifies intent, while enforcing immutability. Same goes for parameters, Once used to it, it is very hard to go back; everything

Reply:
I definitely understand the consistency issue, but I would far rather see us move towards use rather than away. It avoids silly mistakes and helps enshrine immutability as one of the most useful patterns we have.

Reply:
I agree with Bryan; I'm going to leave this as-is.